stistools/mkfringeflat.py

```diff
@@ -0,0 +1,317 @@
+    
+    centera2 = flt_hdulist[0].header['centera2']
+    sizaxis2 = flt_hdulist[0].header['sizaxis2']
+
+    flt_ltv2 = 1 - centera2 - sizaxis2/2
+
+    say0 = int(round(flt_ltv2 - ltv2))
+    say1 = say0 + nrows - 1
+
+    aperture = sci_hdulist[0].header['aperture']
+
+    if extrsize is None:
+        if "0.3X0.09" in aperture:
+            numextrows = int(round(9.0 / binlines + 0.25))
+        elif "0.2X0.06" in aperture:
+            numextrows = int(round(7.0 / binlines + 0.25))
+        elif "52X" not in aperture:
+            print("ERROR: not able to understand APERTURE keyword")
+            return
+        else:
+            numextrows = 11/binlines
+        apername = " [Aperture: " + aperture + "]"
+    else:
+        numextrows = extrsize
+        apername = ""
+
+    if extrloc is None:
+        maxrow = int(round(findloc.find_loc(scidata)))
+    else:
+        maxrow = extrloc
+
+    print(" Extraction center: row {}".format(maxrow))
+    print("   Extraction size: {} pixels {}".format(numextrows, apername))
+    fline = maxrow - int(round((numextrows - 0.49999)/2.0))
+    lline = maxrow + int(round((numextrows - 0.49999)/2.0)) + 1
+
+    if opti_sreg is None:
+        if "G750M" in opt_elem:
+            colstart = int(83/bincols)
+            colstop = int(1106/bincols)
+        else:
+            colstart = int(5/bincols) - 1
+            colstop = int(1020/bincols)
+
+    if rms_region is None:
+        rms_start = int(725/bincols) - 1
+        rms_stop = int(900/bincols)
+
+    print("Range to be normalized: [{}:{},{}:{}]".format(fline, lline, colstart, colstop))
+
+    shifted_flat = None
+
+    if do_shift:
+        print("")
+        print("Determining best shift for fringe flat")
+        print("")
+
+        expo = int(round(math.log10(shift_step)-0.49999999))
+
+        fshift = int(round(beg_shift/(10**(expo))))
+        lshift = int(round(end_shift/(10**(expo))))
+        step = int(round(shift_step/(10**(expo))))
+        flt_blk = block_reduce(fltdata, (binlines/fltbinlines, bincols/fltbincols),
+                                func=np.mean)
+        nshifts = (lshift - fshift)/step + 1
+        
+        rmsvalues = np.zeros(nshifts)
+        current_shift = np.zeros(nshifts)
+
+        for i in range(nshifts):
+            current_shift[i] = beg_shift + i*shift_step
+
+            shifted_flat = shift(flt_blk, current_shift[i], order=1, mode='nearest')
+
+            star_cont_shift = scidata / shifted_flat
+
+            if "G750M" in opt_elem:
+                norm_star_cont_shift = response(star_cont_shift[fline:lline,colstart:colstop],
+                                                star_cont_shift[fline:lline,colstart:colstop],
+                                                threshold=1.0e-20,
+                                                function="spline3",
+                                                sample="*",
+                                                naverage=2,
```

Comment:
`naverage` doesn't get used below, which implies it should be 1 and anything else should raise a `NotImplementedError`.